src/Mod/Part/BasicShapes/SkipGui.py

```diff
@@ -0,0 +1,271 @@
+                sel.Object.isDerivedFrom("Part::FeaturePython") or
+                bool(hasattr(sel.Object,"Shape") and len(sel.Object.Shape.Vertexes) == 1) or
+                sel.Object.isDerivedFrom("Sketcher::SketchObject")) and
+                sel.HasSubObjects == False]
+    if len(selected) != 1:
+        App.Console.PrintLog(f"makeExtrude() need exactly 1 selected profile to skip dialog, not skipping\n")
+        return False
+
+    solid = "True"
+    base = selected[0]
+    if hasattr(base.Shape, "Wire1"):
+        if not base.Shape.Wire1.isClosed():
+            solid = "False"
+    else:
+        if not bool(hasattr(base,"Shape") and len(base.Shape.Vertexes) == 1):
+            App.Console.PrintLog(f"makeExtrude() no wire in shape and not a vertex (not skip dialog)\n")
+            return False #no Wire1
+        else:
+            solid = "False"
+
+    spines = [sel for sel in selx if sel.HasSubObjects]
+    if len(spines) != 1:
+        App.Console.PrintLog(f"makeExtrude() all edges not from same object or no edges selected, not skipping dialog\n")
+        return False
+
+    spineEdges = spines[0].SubElementNames
+    subNames = [name for name in spineEdges if "Edge" in name]
+    if len(subNames) != 1:
+        App.Console.PrintLog(f"makeExtrude() only 1 edge supported for DirLink, not skipping dialog\n")
+        return False
+
+    dirLink = spines[0].Object.getSubObject(subNames[0])
+    if dirLink.Curve.TypeId != "Part::GeomLine":
+        App.Console.PrintLog(f"makeExtrude() selected edge must be of TypeId Part::GeomLine, is TypeId = {dirLink.Curve.TypeId}, not skipping dialog\n")
+        return False
+
+    if solid == "False":
+        App.Console.PrintWarning("Solid = False\n")
+
+    doc = selected[0].Document
+    cmd = getAddObjStr(doc, "Part::Extrusion", "Extrude")
+    actStr = getActStr(doc)
+    cmd += actStr + ".Base = " + getObjStr(doc, selected[0]) + "\n"
+    cmd += actStr + ".Solid = " + solid + "\n"
+    cmd += actStr + ".DirMode = 'Edge'\n"
+    cmd += actStr + ".Reversed = False\n"
+    cmd += actStr + ".Symmetric = False\n"
+    cmd += setHiddenStr(doc, selected)
+    cmd += actStr + ".DirLink = (" + getObjStr(doc, spines[0].Object) + ", '" + subNames[0] + "')\n"
+    Gui.doCommand(cmd)
+    doc.recompute()
+    return True
+
+
+def makeRevolve():
+    selx = Gui.Selection.getSelectionEx()
+    if len(selx) == 0:
+        App.Console.PrintLog(f"makeRevolve() nothing selected, not skipping dialog\n")
+        return False
+
+    selected = [sel.Object for sel in selx if bool(sel.Object.isDerivedFrom("Part::2DObject") or
+                sel.Object.isDerivedFrom("Part::Face") or
+                sel.Object.isDerivedFrom("Part::FeaturePython") or
+                bool(hasattr(sel.Object,"Shape") and len(sel.Object.Shape.Vertexes) == 1) or
+                sel.Object.isDerivedFrom("Sketcher::SketchObject")) and
+                sel.HasSubObjects == False]
+    if len(selected) != 1:
```

Comment:
Does Part/Revolve works if AxisLink isn't a Line but a Curve ?

Reply:
It can be line segment or arc/circle.  In case of a line segment the line segment serves as the axis.  In the case of an arc/circle the center and the normal serve as the axis.  I have added an additional check to ensure these are the curve types selected.

Reply:
Note: Part::GeomLine and Part::GeomCircle are the only 2 types supported.

https://github.com/FreeCAD/FreeCAD/blob/0bbc253ddee6801770c78e011c5201b7c2c36080/src/Mod/Part/App/FeatureRevolution.cpp#L111